src/factorFinder/OutputConfig.java

```diff
@@ -1,4 +1,7 @@
+package com.example.factorfinder;
+import java.util.List;
+
 public interface OutputConfig {
   void writeOutputData(List<String> data);
-  List<String> getOutputData();

```

Comment:
Ok I get why you removed this, we are writing the output data. Same reason why you removed it from the fileoutputconfig.